hatchet/util/rcmanager.py

```
@@ -0,0 +1,150 @@
+                    key
+                )
+            )
+        else:
+            return value
+
+    def list_validator(self, key, value):
+        if not isinstance(value, list):
+            raise TypeError(
+                'Error loading configuration: Configuration "{}" must be of type list'.format(
+                    key
+                )
+            )
+        else:
+            return value
+
+    def dict_validator(self, key, value):
+        if not isinstance(value, dict):
+            raise TypeError(
+                'Error loading configuration: Configuration "{}" must be of type dict'.format(
+                    key
+                )
+            )
+        else:
+            return value
+
+    def validate(self, key, value):
+        try:
+            return self._validations[key](key, value)
+        except TypeError as e:
+            raise e
+        except KeyError:
+            pass
+
+    def _set_validators_to_configs(self):
+        # real configurations
+        self._validations["logging"] = self.bool_validator
+        self._validations["log_directory"] = self.str_validator
+        self._validations["highlight_name"] = self.bool_validator
+        self._validations["invert_colormap"] = self.bool_validator
+
+
+class RcManager(MutableMapping, dict):
+    """
+    A runtime configurations class; modeled after the RcParams class in matplotlib.
+    The benifit of this class over a dictonary is validation of set item and formatting
+    of output.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._validator = ConfigValidator()
+        self.update(*args, **kwargs)
+
+    def __setitem__(self, key, val):
+        """
+        Function loads valid configurations and prints errors for invalid configs.
+        """
+        try:
+            self._validator.validate(key, val)
+            return dict.__setitem__(self, key, val)
+        except TypeError as e:
+            print(e)

```

Do we just want this to print the `TypeError`? I'd normally say we should raise it so that the user's code doesn't continue with invalid settings.

---

Yeah, this is a good point. I was thinking it might be more convenient to a user if an invalid configuration did not shut down the whole program and instead warned the user and defaulted to preloaded configurations. However, I agree they will not want it to run at all if their configurations are incorrect.